### ISSUE TITLE
Step change in Doptimize results for runsize > 500

### ISSUE THREAD

**Issue by GeoMattB** [eendebakpt/oapackage#231]
TLDR: is there something that oapackage.Doptim.Doptimize does when the runsize is >500 compared to a runsize of 500 or lower and if so, how do I get it to give similar results above and below 500.  

EDIT: I did some digging and I see that arrays >500 aren't supported. Still hoping there's a workaround?

![Screenshot 2023-05-13 155754](https://github.com/eendebakpt/oapackage/assets/29564171/6a7fc262-20ee-4868-99a7-12346b587eca) 

I'm attempting to optimize a design / array. For a runsize of 500 and lower  the code takes a lot longer to run and produces more warnings with (depending on other variables) often fewer results. However for a runsize >500 it runs an order of magnitude faster with fewer warnings and always the nrestarts value I've specified is the number of results. I think this is causing some issues with the calculations on the efficiencies that I need when I run experiments with a runsize >500 as the results for the low runsizes don't align with the ones over 500. This occurs for all of my designs.

```
import oapackage
runsize=500  ## test by changing this to 1000 and 400
runtime=1000
factor_levels=[6, 5, 4, 3]
number_of_factors=4
alpha = [5, 5, 15] # optimisation function
nrestarts=10
niter=1000

arrayclass = oapackage.arraydata_t(factor_levels, runsize, 0, number_of_factors)

scores, design_efficiencies, designs, ngenerated = oapackage.Doptim.Doptimize(
            arrayclass,
            nrestarts=nrestarts,
            niter=niter,
            optimfunc=alpha,
            nabort=3000,
            maxtime=runtime)

```
Thanks

**Comment by eendebakpt** [comment]
@GeoMattB To be honest, I never imagined anyone trying to run the `Doptimize` on such large matrices. For runsize > 500 the code is not performing any optimizations (so it returns fast) and you are basically getting random results. The warning should have been an error, I will correct this in the next version of the package.

I will also update the limit of 500 to a larger limit. Memory should not be an issue (as long as the number of columns is reasonable), although I do not know how well the algorithm will perform.

Which warnings do you get for runsize < 500? If you could provide a minimal example with the settings  (runsize, factor levels, number of factors etc.) that would be great

**Comment by GeoMattB** [comment]
awesome, thanks for this. I did some further work after I posted this and the warnings for designs <500 were not coming from OApackage. They were elsewhere in the code I'm using.
I modified DoEgen ( https://github.com/sebhaan/DoEgen ) to use python multiprocessing for the time consuming part (Doptimize) which made it possible to test designs with a runsize of > 500. Though it starts to slow down around there it's useful to demonstrate I've considered larger runsizes.
Thanks very much again.
